### PR TITLE
feat: Create SBOM of container image

### PR DESCRIPTION
**Pull request by viccuad**
## Description

Partial fix of https://github.com/kubewarden/kubewarden-controller/issues/261

As it stands on this commit, the functionality of creating SBOMs added, but never
shipped in a release.

One would need to download them from the `release.yml` worklow and add
them to the release, as done in policy-server.

This isn't as trivial as in policy-server, as we use a 3rd workflow in
`container-build.yml` to trigger `container-image.yml`, and that makes
it difficult (if not impossible) to download them. I have tried several incantations of workflows and failed.

The `container-image.yml` workflow is parametrized to build the
kubewarden-controller container image as a tar, to be consumed by the
e2e tests. This is done so we don't polute the registry with container
images for each branch/PR/main commit. Maybe we should revisit this and
find a different approach.

The `container-image.yml` workflow is being consumed in `e2e-tests.yml`, which passes the output
to
https://github.com/kubewarden/kubewarden-end-to-end-tests/blob/main/.github/workflows/e2e-tests.yml.
Then, it is
passed to kubewarden/setup-kubewarden-cluster-action@v1.1.0.

## Test

To test this pr on your fork,

1. delete fake tag: `git tag -d v1.2.0-rc1 && git push --delete yourremote v1.2.0-rc1`
2. delete old releases manually in your forks' release page
3. push branch to your own fork
2. create fake tag: `git tag -a v1.2.0-rc1 -m "v1.2.0-rc1" -s && git push yourremote main v1.2.0-rc1`
4. watch workflow runs

Example of a workflow run with SBOM artifacts: https://github.com/viccuad/kubewarden-controller/actions/runs/3023713161

### Potential improvement

Missing:
- Depend on ci.yml for workflows that should
- Currently we only sign production images. But the way we do it via `container-build.yml` is clumsy. Maybe it should be a conditional on `container-image.yml` when triggered by a tag.
- Remove `container-build.yml`, only use `container-image.yml`. Depend on `container-image.yml` for `e2e-tests.yml`.
- The tricky one: Release should depend on `container-image.yml` succeeding (so we have the SBOM artifacts in GHA) with something like:
  ```
  name: Release kubewarden-controller
  on:
    workflow_run:
      workflows: ["Build container image"]
      types:
        - completed
      branches:
        - "main"
    push:
      tags:
      - 'v*'

  jobs:
   release:
      name: Create release
      runs-on: ubuntu-latest
      if: ${{ github.event.workflow_run.conclusion == 'success' }}
  ```
  But then, I can't get release to trigger on tags..

### CODE REVIEW

**Review by raulcabello — COMMENTED** [review]
perhaps I'm missing something, but why do we want to add the container image sbom file to the release? We already attach (and sign) the sbom to the OCI registry with cosign. Users can retrieve sbom with `cosign download sbom`, this will also verify sbom was properly signed.

File: .github/workflows/container-image.yml
Comment (raulcabello) [inline .github/workflows/container-image.yml:180]: we don't sign the sbom we attach. We should probably sign it after this or use in-toto attestations
more info https://docs.sigstore.dev/cosign/other_types/#sboms-software-bill-of-materials

**Comment by raulcabello** [comment]
It is very complicated to add the sbom to the release with the current approach as @viccuad mentioned. I run the container creation inside of the release workflow [here](https://github.com/kubewarden/kubewarden-controller/pull/276) then it is easy to add the sbom to the release.
I think it make sense to run the container-build inside of the release workflow, so we don't create a release until the container is created.

File: .github/workflows/container-image.yml
Comment (raulcabello) [reply .github/workflows/container-image.yml:180]: I saw we are signing it in a different step, sorry about the noise

**Review by flavio — APPROVED** [review]
LGTM, waiting for @viccuad to give his approval once he's back from vacation

**Review by flavio — CHANGES_REQUESTED** [review]
I take my approval back, https://github.com/kubewarden/kubewarden-controller/pull/276 seems a better solution

**Comment by viccuad** [comment]
Superseded by https://github.com/kubewarden/kubewarden-controller/pull/276.

**Pull request closed without merging**
